Sources/Internal/WebSocketConnection.swift

```diff
@@ -3,120 +3,303 @@
 //
 
 import Foundation
-import Starscream
 import Network
 
+enum WebSocketError: Error {
+    case closedUnexpectedly
+    case peerDisconnected
+}
+
 class WebSocketConnection {
     let url: WCURL
-    private let socket: WebSocket
-    
     private var isConnected: Bool = false
-    
+    private var task: URLSessionWebSocketTask?
+    private lazy var session: URLSession = {
+        let delegate = WebSocketConnectionDelegate(eventHandler: { [weak self] event in
+            self?.handleEvent(event)
+        })
+        let configuration = URLSessionConfiguration.default
+        configuration.shouldUseExtendedBackgroundIdleMode = true
+        configuration.waitsForConnectivity = true
+
+        return URLSession(configuration: configuration, delegate: delegate, delegateQueue: nil)
+    }()
+
+#if os(iOS)
```

Comment:
I'm using `#if os(iOS)` just because I saw it in #81 but I wonder, is it really necessary? Doesn't look like other platforms are explicitly supported (beyond M1 macs running iOS apps)?

Reply:
Package's platform include macOS and the `UIBackgroundTaskIdentifier` does not exist on macOS, so I thing we're fine here